### DISCUSSION TITLE
makefile  and temporary table for rpgle compilation

### DISCUSSION THREAD

**Discussion by pabloto** [codefori/vscode-ibmi#522] [IBM i] [answered]
Hi, 

I have an Rpgle than use a file in qtemp, when it run before call the rpgle a clle create that file, usually when I have to compile it, I create that in qtemp than I make my compile. 

Trying to put into a makefile I tried with different approch, but no one works, I think my problem is relative to the thread where my command run.

Here are some example:
BIN_LIB=MYLIB
SHELL=/QOpenSys/usr/bin/qsh

all: my.rpgle 

%.rpgle:
	system -s "chgcurlib curlib($(BIN_LIB))";\
	system -vi "RUNSQLSTM SRCSTMF('./qfdssrc/workfile.sql') COMMIT(*NONE)";\
	system -s "CRTBNDRPG PGM($(BIN_LIB)/$*) SRCSTMF('./qrpglesrc/$*.rpgle') OPTION(*EVENTF) TGTCCSID(*JOB)";

the workfile.sql contain the create table in qtemp

or:

%.rpgle:
	system -s "chgcurlib curlib($(BIN_LIB))";\
	db2util "create table qtemp/mytable (field1 char(10)) Rcdfmt mytabler;\
	system -s "CRTBNDRPG PGM($(BIN_LIB)/$*) SRCSTMF('./qrpglesrc/$*.rpgle') OPTION(*EVENTF) TGTCCSID(*JOB)";

the same with db2 instead of db2util...

I always get a compile error about relative to my work file doesn't exit.
Any idea?

Many thanks

**Comment by worksofliam** [comment, marked as answer]
The system command generally runs in a new job, so the user of `QTEMP` is really possible in this case. You build a temporarily library for the build and then delete it when you're done?

**Reply by pabloto** [reply]
A bit rude but it can be work, I'll try.

**Reply by worksofliam** [reply]
Post back what you try! I am curious! 😄

**Reply by pabloto** [reply]
@worksofliam it works. 


BIN_LIB=MYBINLIB
LIBLIST=MYLIBRARYLIST
SHELL=/QOpenSys/pkgs/bin/bash
ASP=/MYASP/QSYS.LIB

all: myrpg.rpgle mypgm2.rpgle

%.rpgle:
	system -s "crtlib lib(forcompile) aspdev(directa1)";
	db2util "create Table forcompile.workfile (Logrec Char(512)) Rcdfmt workrec";
	liblist -a forcompile;\\
	system -s "CRTBNDRPG PGM($(BIN_LIB)/$*) SRCSTMF('./qrpglesrc/$*.rpgle') OPTION(*EVENTF) TGTCCSID(*JOB)";
	system -s "dltlib lib(forcompile)";

**Reply by worksofliam** [reply]
@pabloto Great work
